### ISSUE TITLE
Codespell report for "OpenSSL" (on fossies.org)

### ISSUE THREAD

**Issue by jschleus** [openssl/openssl#10998]
The FOSS server [fossies.org](https://fossies.org) - also supporting "OpenSSL"  - offers a new feature "[Source code misspelling reports](https://fossies.org/features.html#codespell). Such reports are normally only generated on request, but as Fossies administrator I have just created for testing purposes such a [codespell](https://github.com/codespell-project/codespell) based analysis for the `openssl` release "1.1.1d":

 https://fossies.org/linux/misc/openssl/codespell.html

That version-independent (not linked) URL should always redirect to the last report (if available), so currently to

 https://fossies.org/linux/misc/openssl-1.1.1d.tar.gz/codespell.html

Although after a first review some obviously wrong matches ("False Positives" = FPs) are already filtered out (ignored) please inform me if you find more of them so that I can force a new improved check if applicable.

Just for information there are also two supplemental pages https://fossies.org/linux/misc/openssl/codespell_conf.html (showing some used "codespell" configurations) and https://fossies.org/linux/misc/openssl/codespell_fps.html (showing all resulting obvious "False Positives").

If  appropriate, additional reports for the current release or also master versions (these in a
special "test" folder) could be willingly created.

A remark: As a very new feature in the report page there was now added a one-letter column (also sortable), which should indicate the error context (mouse over optionally expands the letter information). This may make it easier to identify the potentially more serious errors, especially since additionally the letters for errors that are "only" in comments (`c` and `C`) and pure text files (`t`) are greyed out. The errors marked by a `?` may be in the source code itself. As C-layman I was unsure if the both spelling errors of that type `bion` (suggestion `biom`?) and `interm` (suggestion  `intern`?) are really spelling errors or FPs.

**Comment by mspncp** [comment]
Cool, thanks for pointing us to the link! We will take a look at it.

BTW: Is your spell checker tool also able to generate a patch for fixing all those typos? ;-)

**Comment by richsalz** [comment]
Neat.  The project has taken the view that the pre-Apache license is sacrosanct and that not a character (even typo's) of it will be touched.

Can you run this against master?  That's the version most likely to effect change.

**Comment by richsalz** [comment]
If there a way to "turn off" some things, like publicly/publically?

**Comment by mspncp** [comment]
See #10961 
_(The file [codespell.openssl-1.1.1d.txt](https://fossies.org/linux/misc/openssl-1.1.1d.tar.gz/codespell.openssl-1.1.1d.txt) was ideal input for Emacs' compilation mode; I just couldn't resist to do it...)_

**Comment by mspncp** [comment]
> publicly/publically?

dict.cc says, the latter is rarely used, so I replaced it. I could undo that if you prefer.

**Comment by mspncp** [comment]
> The project has taken the view that the pre-Apache license is sacrosanct

I left it untouched in #11000.

**Comment by jschleus** [comment]
@mspncp: Fossies currently doesn't offer patch files. Some more infos  you may found [here](https://fossies.org/features.html#codespell) under the paragraph beginning with "A note:" (see also the coy tooltip hint if your mouse is over the `*)` at the end of that paragraph). By the way I'm trying to write a bash script that will do some according tasks but that has just only alpha quality and isn't suited for a project like OpenSSL.

@richsalz: Yes, single words, files or even sub-directories can be ignored (excluded). So for now just the contested word `poping` is also ignored (since it's as mentioned  in an according  pull request related to `pop` and the suggestion `popping` seems false).

As requested a  codespell report for the current master was now generated in a special "test" folder (that isn't integrated in the Fossies standard services and should - at least principally - not accessible by search engines):

 https://fossies.org/linux/test/openssl-master.tar.gz/codespell.html

That URL is currently an alias for

 https://fossies.org/linux/test/openssl-master.7f293d9.200202.tar.gz/codespell.html

I refrained from ignoring the `License` file since in master there seems a new `License` file in use that has been so also checked (without errors). Also the file `CHANGES` is currently not yet ignored (although the historical spelling errors will probably remain unchanged).

By the way, the (not very relevant) Fossies ratings "A-" (for 1.1.1d) and "B+" (for master 7f293d9) are astonishingly good.

And just for fun I generated also a so-called Fossies "Diffs"-report [1.1.1d_vs_7f293d9.200202](https://fossies.org/diffs/test/openssl/1.1.1d_vs_7f293d9.200202/) between the latest release and the current master. Wow, there seem to be a lot of changes!

**Comment by mattcaswell** [comment]
> By the way, the (not very relevant) Fossies ratings "A-" (for 1.1.1d) and "B+" (for master 7f293d9) are astonishingly good.

It's not the first time people have run spell checkers on the OpenSSL source ;-)

**Comment by mspncp** [comment]
> So for now just the contested word `poping` is also ignored (since it's as mentioned in an according pull request related to `pop` and the suggestion `popping` seems false).

FWIW: this refers to my https://github.com/openssl/openssl/pull/11000#commitcomment-37082260, which for some reason was not properly attached  to [this line of code](https://github.com/openssl/openssl/pull/11000/files#diff-318ae77134d5eb0a20b9c26c5bda9af8R38).

**Comment by mspncp** [comment]
> There is also a conflict in rand_lcl.h

The conflict was caused by the fact that I based my topic branch onto the tag OpenSSL_1_1_1d to get the same line numbers as in [codespell.openssl-1.1.1d.txt](https://fossies.org/linux/misc/openssl-1.1.1d.tar.gz/codespell.openssl-1.1.1d.txt) and forgot to rebase it to the tip of OpenSSL_1_1_1-stable afterwards. I just pushed a rebased version (without other changes).

Apropos rebase: git had no problems to detect the renaming `crypto/rand_rand_lcl.h` -> `crypto/rand_rand_local.h`. Maybe GitHub's rename limits are set too low.

```
$ git config -l | grep limit
merge.renamelimit=32768
diff.renamelimit=32768
```

**Comment by mspncp** [comment]
Ups, wrong thread.

**Comment by richsalz** [comment]
@jschleus thanks again for this.  For what it's worth, I think it would be most useful to the OpenSSL project if master were regularly checked.  If doing "git clone" doesn't work for you, you can find daily snapshots at https://www.openssl.org/source/snapshot/

**Comment by jschleus** [comment]
@richsalz First I tried the - at least for me - simpler solution using the daily snapshots but I wasn't really satisfied (also because the spelling reports would be lag behind the current source code til a day). Since I'm not a Git expert I've gone some wrong ways but now the "Git"-solution seems to work:

 https://fossies.org/linux/test/openssl-master.tar.gz/codespell.html

and also

 https://fossies.org/linux/test/openssl-1.1.1-snap.tar.gz/codespell.html

should always show the respective codespell reports of the current version (the time delay should be currently max. ten minutes).

**Comment by mspncp** [comment]
Fixed, since #11000 has been merged. Thank you again @jschleus for doing the spellcheck!
